reference/Element.cpp

```diff
@@ -0,0 +1,163 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+   Copyright 2022 The StableHLO Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "Element.h"
+
+#include <complex>
+
+#include "llvm/ADT/APFloat.h"
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/BuiltinTypes.h"
+
+namespace mlir {
+namespace stablehlo {
+
+namespace {
+
+bool areApproximatelyEqual(APFloat f, APFloat g) {
+  llvm::APFloatBase::cmpResult cmpResult = f.compare(g);
+  if (cmpResult == APFloat::cmpEqual) return true;
+
+  if (cmpResult == APFloat::cmpUnordered) return f.isNaN() == g.isNaN();
+
+  auto absFloat = abs(f - g);
+  APFloat err(f.getSemantics(), "1e-5");
+  return absFloat.compare(err) == APFloat::cmpLessThan;
+}
+
+bool isSupportedUnsignedIntegerType(Type type) {
+  return type.isUnsignedInteger(4) || type.isUnsignedInteger(8) ||
+         type.isUnsignedInteger(16) || type.isUnsignedInteger(32) ||
+         type.isUnsignedInteger(64);
+}
+
+// TODO(https://github.com/openxla/stablehlo/issues/22): Support signed integers
```

Comment:
Also, can you leave a slightly longer comment that explains why we currently map signless integers on signed integers?